src/cluster.h

```diff
@@ -330,6 +330,40 @@ typedef struct {
     union clusterMsgData data;
 } clusterMsg;
 
+/* clusterMsg defines the gossip wire protocol exchanged among Redis cluster
+ * members, which can be running different versions of redis-server bits,
+ * especially during cluster rolling upgrades.
+ *
+ * Therefore, fields in this struct should remain at the same offset from
+ * release to release. The static asserts below ensures that incompatible
+ * changes in clusterMsg be caught at compile time.
+ */
+
+/* Avoid static_assert on non-C11 compilers. */
+#if __STDC_VERSION__ >= 201112L
```

Comment:
Isn't this macro affected by -std=c99 so these asserts will never be checked?

Reply:
@zuiderkwast It is affected, but why would they never be checked? We use `-std=c11` to get C11 atomics on relevant platforms, we just don't adopt C11 all over (yet).

Reply:
OK, I though we use c99 everyehere. If we use c11 in at least one build, we'll catch it. :dart:

Reply:
I can add an implementation for static_assert for pre-C11 to make this behavior consistent across all supported compilers.  thoughts?

Reply:
@PingXie I don't think it is necessary, since as long as we have at least one C11 compiler in CI (and we do), we're covered.

Reply:
Sure. It is actually pretty trivial (3 lines) but yeah agreed that there is no urgency. 

```
#ifndef static_assert
#define static_assert(expr, lit) extern char __static_assert_failure[(expr) ? 1:-1]
#endif
```
PS, to your earlier question, the string literal is no longer required in [C23](https://en.cppreference.com/w/c/language/_Static_assert) but still needed for C11, which is what I meant by "old compliers", relatively speaking :).